whatsapp_theming/lib/main.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:flutter/material.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+import 'darkmode.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:theme_provider/theme_provider.dart';
+import 'package:flutter/scheduler.dart';
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatefulWidget {
+  @override
+  _MyAppState createState() => _MyAppState();
+}
+
+class _MyAppState extends State<MyApp> {
+  int radiovalue;
+  loadRadiovalue() async {
+    SharedPreferences prefs = await SharedPreferences.getInstance();
+    dynamic result = prefs.getInt('radiovalue');
+    if (result != null) {
+      return result;
+    } else {
+      return 2;
+    }
+  }
+
+  @override
+  void initState() {
+    loadRadiovalue().then((val) {
+      setState(() {
+        radiovalue = val;
+      });
+    });
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
```

Comment:
Split your code into parts. Write code comments. This repo is to showcase code that is easy to consume for beginners. This is a very complex way of writing code.